llvm/lib/CodeGen/GlobalISel/CallLowering.cpp

```diff
@@ -845,30 +897,48 @@ bool CallLowering::handleAssignments(ValueHandler &Handler,
                                      DstMPO, DstAlign, SrcMPO, SrcAlign,
                                      MemSize, VA);
         }
-        continue;
-      }
-
-      assert(!VA.needsCustom() && "custom loc should have been handled already");
-
-      if (i == 0 && !ThisReturnRegs.empty() &&
-          Handler.isIncomingArgumentHandler() &&
-          isTypeIsValidForThisReturn(ValVT)) {
+      } else if (i == 0 && !ThisReturnRegs.empty() &&
+                 Handler.isIncomingArgumentHandler() &&
+                 isTypeIsValidForThisReturn(ValVT)) {
         Handler.assignValueToReg(ArgReg, ThisReturnRegs[Part], VA);
-        continue;
-      }
-
-      if (Handler.isIncomingArgumentHandler())
+      } else if (Handler.isIncomingArgumentHandler()) {
         Handler.assignValueToReg(ArgReg, VA.getLocReg(), VA);
-      else {
+      } else {
         DelayedOutgoingRegAssignments.emplace_back([=, &Handler]() {
           Handler.assignValueToReg(ArgReg, VA.getLocReg(), VA);
         });
       }
+
+      // Finish the handling of indirect parameter passing when receiving
+      // the value (we are in the called function or the caller when receiving
+      // the return value).
+      if (VA.getLocInfo() == CCValAssign::Indirect &&
+          Handler.isIncomingArgumentHandler()) {
+        Align Alignment = DL.getABITypeAlign(Args[i].Ty);
+        MachinePointerInfo MPO;
+
+        if (VA.isMemLoc())
+          MPO = MachinePointerInfo::getStack(MF, VA.getLocMemOffset());
```

Comment:
I'd suggest to replace this with
```
MachinePointerInfo MPO = MachinePointerInfo::getUnknownStack(MF)
```
for the reason outlined in the other comment (we don't know the exact location we're loading from).

Reply:
It seems to solve the issue with loads right?

Reply:
@michaelmaitland 
Or, should we create the info with the IR object instead?... My knowledge in this area is vague.

Reply:
> It seems to solve the issue with loads right?

It did. At least the info is now not wrong :)

Reply:
I am sorry, but I am not sure the correct answer.